sdk/nodejs/provider/experimental/schema.ts

```diff
@@ -65,21 +65,36 @@ export interface PackageSpec {
     name: string;
     version?: string;
     description?: string;
+    namespace?: string;
     resources: { [key: string]: Resource };
     types: { [key: string]: ComplexType };
     language?: { [key: string]: any };
 }
 
 export function generateSchema(
     packageJSON: Record<string, any>,
+    pulumiYAML: Record<string, any>,
     components: Record<string, ComponentDefinition>,
     typeDefinitions: Record<string, TypeDefinition>,
     packageReferences: Record<string, string>,
 ): PackageSpec {
-    const providerName = packageJSON.name;
+    const matches = packageJSON.name.match(/(@.*?\/)?(.+)/);
+    let providerName = matches[2];
+    let namespace = undefined;
+    if (matches[1]) {
+        namespace = matches[1].substring(1, matches[1].length - 1);
+    }
+    // Setting the namespace or name in PulumiPlugin.yaml overrides the ones from package.json
+    if (pulumiYAML["namespace"]) {
+        namespace = pulumiYAML["namespace"];
```

Comment:
Oh, do we have to do PulumiPlugin.yaml parsing in each SDK separately? I imagined that would happen centrally when loading the plugin in the engine, that would override the namespace property of the schema?

Reply:
I *think* we have to.  The `GetSchema` request is handled directly by the provider, so I don't think we have a way to inject anything there.  We could maybe fix it up at all the callsites, but that sounds way messier. If anyone has smart ideas I'd be all ears.

Reply:
Isn't it similar to what we do with the version assignment from git tags?

Reply:
Hrm true, we probably could.  I guess it means we still have to parse it at least in python to get the name, but otherwise we can override it.